commit a543ddfd4ab01c2ab65d419d06f097af1d0ac1f1
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Mar 23 20:53:47 2016 +0000

    vchiq_arm: Service callbacks must not fail
    
    Service callbacks are not allowed to return an error. The internal callback
    that delivers events and messages to user tasks does not enqueue them if
    the service is closing, but this is not an error and should not be
    reported as such.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
index d8669fa..54552c6 100644
--- a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
+++ b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
@@ -224,7 +224,7 @@ add_completion(VCHIQ_INSTANCE_T instance, VCHIQ_REASON_T reason,
 		} else if (instance->closing) {
 			vchiq_log_info(vchiq_arm_log_level,
 				"service_callback closing");
-			return VCHIQ_ERROR;
+			return VCHIQ_SUCCESS;
 		}
 		DEBUG_TRACE(SERVICE_CALLBACK_LINE);
 	}
